ljbuild.make

```diff
@@ -0,0 +1,92 @@
+.SECONDEXPANSION:
+.SUFFIXES:
+
+ifndef name
+$(error build name is not set)
+endif
+
+BUILD_DIR:= builds/$(name)
+BUILD_FULLDIR:= $(shell cd builds && pwd)/$(name)
+BUILD_DIRS= $(BUILD_DIR) $(BUILD_DIR)/include $(BUILD_DIR)/jit $(BUILD_DIR)/jitlog
+REPO:= $(BUILD_FULLDIR)/src
+SRC:= $(BUILD_DIR)/src/src
+
+# We need a branch name when first checking out the repo
+ifndef branch
+  $(if $(shell if [ -f $(REPO)/.git ]; then echo ok; fi), , $(error No 'branch' variable set))
+endif
+
+# Make sure all output build directories are created before we run anything
+$(info $(shell mkdir -p $(BUILD_DIRS)))
+
+SRC_C:= $(wildcard $(SRC)/*.c)
+SRC_H:= $(wildcard $(SRC)/*.h)
+SRC_JITLOG:= $(wildcard $(SRC)/jitlog/*.lua)
+FILES_INC:= luaconf.h lua.h luajit.h lualib.h lauxlib.h
+SRC_INC:= $(addprefix $(SRC)/,$(FILES_INC))
+INCLUDES:= $(addprefix $(BUILD_DIR)/include/,$(FILES_INC))
+
+GITCMD:= git --git-dir="$(REPO)/.git" --work-tree="$(REPO)"
+GITCMD_MAINREPO:= git --git-dir="luajit_repo/.git" --work-tree="luajit_repo"
+
+$(REPO)/.git:
+	mkdir -p $(REPO)
+	(cd luajit_repo && git worktree add $(REPO))
```

Comment:
I don't think brackets are needed here?

Reply:
I was just running it as a sub shell because it was changing a directory to run the command. I just didn't want the current directory to be left changed if command failed.

Reply:
I think each line isolates the CWD anyway?

Reply:
It looks like, i guess i assumed wrong

Reply:
fixed